pallets/domains/src/lib.rs

```diff
@@ -180,23 +198,22 @@ pub mod pallet {
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
-        /// Registers a domain ([full_domain]) using origin with [content],
-        /// and set the domain to expire in [expires_in] number of blocks.
+        /// Registers a domain ([full_domain]) using origin, and set
+        /// the domain to expire in [expires_in] number of blocks.
```

Comment:
it will stay available and anyone could register it again.

But this could be an issue if user A registers the domain and set a record, then the domain expired. User A can no longer retrieve his tokens back.

To solve this I would require to store the account that was reserved, so I can refund it back when it changes.